e2e/__test__/tours.test.js

```diff
@@ -0,0 +1,186 @@
+            "Test3",
+          ],
+          "launchDate": Any<String>,
+          "stops": Array [
+            Object {
+              "_id": Any<String>,
+            },
+          ],
+          "title": "Testing Tour",
+        }
+      `
+      );
+    });
+  });
+
+  it('gets all tours', () => {
+    return Promise.all([
+      postTour(testTour),
+      postTour(testTour),
+      postTour(testTour)
+    ])
+      .then(() => {
+        return request.get('/api/tours').expect(200);
+      })
+      .then(({ body }) => {
+        expect(body.length).toBe(3);
+        expect(body[0]).toMatchInlineSnapshot(
+          {
+            _id: expect.any(String),
+            launchDate: expect.any(String),
+            stops: [
+              {
+                _id: expect.any(String)
+              }
+            ]
+          },
+          `
+          Object {
+            "__v": 0,
+            "_id": Any<String>,
+            "activities": Array [
+              "Test1",
+              "Test2",
+              "Test3",
+            ],
+            "launchDate": Any<String>,
+            "stops": Array [
+              Object {
+                "_id": Any<String>,
+              },
+            ],
+            "title": "Testing Tour",
+          }
+        `
+        );
+      });
+  });
+
+  it('adds a stop to the tour', () => {
+    return postTour(testTour)
+      .then(tour => {
+        return request
+          .post(`/api/tours/${tour._id}/stops`)
+          .send(testLocation)
+          .expect(200)
+          .then(body => {
+            return [body, testLocation, testLocation];
+          });
+      })
+      .then(out => {
+        const stop = out[0].body[1];
+        expect(stop).toMatchInlineSnapshot(
+          {
+            _id: expect.any(String)
+          },
+          `
+          Object {
+            "_id": Any<String>,
+          }
+        `
+        );
+      });
+  });
+
+  it('updates stop attendance', () => {
+    return postTour(testTour)
+      .then(tour => {
+        return request
+          .post(`/api/tours/${tour._id}/stops`)
+          .send(testLocation)
+          .expect(200)
+          .then(out => {
+            const stops = out.body[0];
+            return request
+              .put(`/api/tours/${tour._id}/stops/${stops._id}/attendance`)
+              .send(testAttendance)
+              .expect(200)
+              .then(({ body }) => {
+                expect(body[0].attendance).toBe(24);
+              });
+          });
+      });
+  });
+
+  it('deletes a stop', () => {
+    return postTour(testTour)
+      .then(tour => {
+        return request
+          .post(`/api/tours/${tour._id}/stops`)
+          .send(testLocation)
+          .expect(200)
+          .then(out => {
+            const stops = out.body[1];
+            return request
+              .delete(`/api/tours/${tour._id}/stops/${stops._id}`)
+              .send(tour._id, stops._id)
+              .expect(200);
```

Comment:
what does deleting a stop return? you should test that the return value matches what you expect or check what the tour looks like to show that the stop was actually deleted.